music21/humdrum/spineParser.py

```diff
@@ -2118,7 +2121,85 @@ def getAllOccurring(self):
         return retEvents
 
 
-def hdStringToNote(contents):
+def _hdStringToDuration(contents: str,
+            defaultDuration: duration.Duration|None = None) -> duration.Duration:
```

Comment:
lint -- inconsistent indentation.

I still don't understand why a routine called XToDuration would take in a Duration object?  This does not look right or make sense to me.

Reply:
Looking back at my pr description, it looks like I didn't copy over my previous example. The issue is that you occasionally get a humdrum chord in one voice that only notates the duration of the first note, as in:
8C E G
Humdrum at least allows for this and the Humdrum Verovio Viewer would parse these three notes all as eighth notes. Here's a [minimal example](https://verovio.humdrum.org/?t=KiprZXJuCjRnIDRiCjRmIGEKMmUgZwoxZCBmCj09CiotCg==) showing this behavior. music21 currently parses the C as an eighth note but the E and the G are seen as not having a duration so they get the default duration of a duration object, which is a quarter note. A quarter note is a reasonable default duration to have in music21 but not in this circumstance. In this case it is helpful to be able to say "in humdrum chords, use the first note's duration as the default duration for the other notes in the chord if they don't specify one themselves". This is just like what music21 does for the chord object itself. The chord object not only has the same duration as the first note, it uses the same literal duration object. So this pr extends that same behavior to the other notes in the chord if they don't specify their own duration. So 8C E G would be a chord of three eighth notes but 8C 2E 2G would be a chord of one eighth note and two half notes. I encountered this issue in a Joplin rag kern file, but I don't know how common it is in general. Tests are also added to check this new behavior.